services/navigation.js

```diff
@@ -268,65 +295,107 @@ module.exports = {
+            const parentPath = isExternal
+              ? undefined
+              : `${path === "/" ? "" : path}/${
+                item.path === "/" ? "" : item.path
+              }`;
+            const slug = isString(parentPath)
+              ? slugify(
+                (first(parentPath) === "/"
+                    ? parentPath.substring(1)
+                    : parentPath
+                ).replace(/\//g, "-")
+              )
+              : undefined;
             const lastRelated = item.related ? last(item.related) : undefined;
             return {
-              title: utilsFunctions.composeItemTitle(item, contentTypesNameFields, contentTypes),
+              title: utilsFunctions.composeItemTitle(
+                item,
+                contentTypesNameFields,
+                contentTypes
+              ),
               menuAttached: item.menuAttached,
               path: isExternal ? item.externalPath : parentPath,
               type: item.type,
               uiRouterKey: item.uiRouterKey,
-              slug: !slug && item.uiRouterKey ? slugify(item.uiRouterKey) : slug,
+              slug:
+                !slug && item.uiRouterKey ? slugify(item.uiRouterKey) : slug,
               external: isExternal,
-              related: isExternal || !lastRelated ? undefined : {
-                ...lastRelated,
-                __templateName: getTemplateName(lastRelated.__contentType, lastRelated.id),
-              },
-              audience: !isEmpty(item.audience) ? item.audience.map(aItem => aItem.key) : undefined,
-              items: isExternal ? undefined : service.renderTree({
-                items,
-                id: item.id,
-                field,
-                path: parentPath,
-                itemParser,
-              }),
+              related:
+                isExternal || !lastRelated
+                  ? undefined
+                  : {
+                    ...lastRelated,
+                    __templateName: getTemplateName(
+                      lastRelated.__contentType,
+                      lastRelated.id
+                    ),
+                  },
+              audience: !isEmpty(item.audience)
+                ? item.audience.map((aItem) => aItem.key)
+                : undefined,
+              items: isExternal
+                ? undefined
+                : service.renderTree({
+                  items,
+                  id: item.id,
+                  field,
+                  path: parentPath,
+                  itemParser,
+                }),
             };
           };
+
           const treeStructure = service.renderTree({
             items,
-            field: 'parent',
+            field: "parent",
             itemParser,
           });
+
+          const filteredStructure = filter
+            ? treeStructure.filter((item) => {
+              return item.uiRouterKey === filter;
+            })
+            : treeStructure;
+
```

Comment:
only these changes in this method are new, the rest looks like a problem with code formatting, could you revert this?